pkg/trace/api/fuzz_test.go

```diff
@@ -0,0 +1,237 @@
+
+func commonMsgpackEncoder(traces pb.Traces) ([]byte, error) {
+	return traces.MarshalMsg(nil)
+}
+
+func commonJSONEncoder(traces pb.Traces) ([]byte, error) {
+	return json.Marshal(traces)
+}
+
+type decoder func([]byte) error
+
+func commonMsgpackDecoder(traces []byte) error {
+	var pbTraces pb.Traces
+	_, err := pbTraces.UnmarshalMsg(traces)
+	return err
+}
+
+func commonJSONDecoder(traces []byte) error {
+	var pbTraces pb.Traces
+	return json.NewDecoder(bytes.NewReader(traces)).Decode(&pbTraces)
+}
+
+func FuzzHandleTracesV07(f *testing.F) {
+	decode := func(traces []byte) error {
+		var payload pb.TracerPayload
+		_, err := payload.UnmarshalMsg(traces)
+		return err
+	}
+	fuzzTracesAPI(f, V07, "application/msgpack", commonMsgpackEncoder, decode)
+}
+
+func FuzzHandleTracesV05(f *testing.F) {
+	decode := func(traces []byte) error {
+		var pbTraces pb.Traces
+		return pbTraces.UnmarshalMsgDictionary(traces)
+	}
+	fuzzTracesAPI(f, v05, "application/msgpack", commonMsgpackEncoder, decode)
+}
+
+func FuzzHandleTracesV04Msgpack(f *testing.F) {
+	fuzzTracesAPI(f, v04, "application/msgpack", commonMsgpackEncoder, commonMsgpackDecoder)
+}
+
+func FuzzHandleTracesV04JSON(f *testing.F) {
+	fuzzTracesAPI(f, v04, "application/json", commonJSONEncoder, commonJSONDecoder)
+}
+
+func FuzzHandleTracesV03Msgpack(f *testing.F) {
+	fuzzTracesAPI(f, v03, "application/msgpack", commonMsgpackEncoder, commonMsgpackDecoder)
+}
+
+func FuzzHandleTracesV03JSON(f *testing.F) {
+	fuzzTracesAPI(f, v03, "application/json", commonJSONEncoder, commonJSONDecoder)
+}
+
+func FuzzHandleTracesV02JSON(f *testing.F) {
+	fuzzTracesAPI(f, v02, "application/json", commonJSONEncoder, commonJSONDecoder)
+}
+
+// fuzzTracesAPI allows fuzzing multiple trace APIs.
+// The caller has to provide the targeted API version, the content type and the functions to encode and decode the payload.
+func fuzzTracesAPI(f *testing.F, v Version, contentType string, encode encoder, decode decoder) {
+	conf := newTestReceiverConfig()
+	receiver := newTestReceiverFromConfig(conf)
+	handlerFunc := receiver.handleWithVersion(v, receiver.handleTraces)
+	server := httptest.NewServer(handlerFunc)
+	defer server.Close()
+	for _, n := range []int{1, 10, 100} {
+		pbTraces := testutil.GetTestTraces(n, n, true)
+		traces, err := encode(pbTraces)
+		if err != nil {
+			f.Fatal("Couldn't generate seed corpus:", err)
+		}
+		f.Add(traces)
+	}
+	f.Fuzz(func(t *testing.T, traces []byte) {
+		req, err := http.NewRequest("POST", server.URL, bytes.NewReader(traces))
+		if err != nil {
+			t.Fatal("Couldn't create http request:", err)
+		}
+		req.Header.Set("Content-Type", contentType)
+		var client http.Client
+		resp, err := client.Do(req)
+		if err != nil {
+			// Most likely caused by a network issue (out of scope)
+			t.Skip("Couldn't perform http request:", err)
+		}
+		defer func() {
+			if err = resp.Body.Close(); err != nil {
+				t.Log("Couldn't close response body:", err)
```

Comment:
I'd probably Error instead of Log here, since the log won't show up unless its in a failure state or if `-v` is set.